include/pybind11/pybind11.h

```diff
@@ -2770,7 +2776,12 @@ get_type_override(const void *this_ptr, const type_info *this_type, const char *
         PyCodeObject *f_code = PyFrame_GetCode(frame);
         // f_code is guaranteed to not be NULL
         if ((std::string) str(f_code->co_name) == name && f_code->co_argcount > 0) {
+#        if PY_VERSION_HEX >= 0x030d0000
+            PyObject *locals = PyEval_GetFrameLocals();
+#        else
             PyObject *locals = PyEval_GetLocals();
+            Py_INCREF(locals);
```

Comment:
@henryiii This `Py_INCREF(locals)` appears before the `locals != nullptr` test, therefore I think it could therefore segfault in object.h: `op->ob_refcnt++;`. Did you consider that already? Did you mean to use `Py_XINCREF(locals)` here?

Reply:
Yes I did, thanks! I think it’s very unlikely that this will be null, but best to be correct!